rasa/core/events/__init__.py

```diff
@@ -245,18 +247,23 @@ def empty():
         return UserUttered(None)
 
     def as_dict(self):
-        d = super(UserUttered, self).as_dict()
-        input_channel = None  # for backwards compatibility (persisted evemts)
+        useruttered_dict = super(UserUttered, self).as_dict()
+        input_channel = None  # for backwards compatibility (persisted events)
+        metadata = None  # for backwards compatibility (persisted events)
         if hasattr(self, "input_channel"):
             input_channel = self.input_channel
-        d.update(
+        if hasattr(self, "metadata"):
```

Comment:
I'm not sure, because the `message_id` comes from `self`:
https://github.com/RasaHQ/rasa/blob/7c9fd15a6f2b3e393d8a9359eb3934195b64b8d9/rasa/core/events/__init__.py#L262
:thinking:

Reply:
> I'm not sure, because the message_id comes from self:

So does `self.metadata`, right?

We could also use `getattr` and avoid all these `if`s. e.g.

```python
{
  "metadata": getattr(self, "metadata", None)
}